src/test/kotlin/lotto/domain/PurchaseSessionTest.kt

```diff
@@ -0,0 +1,106 @@
+package lotto.domain
+
+import lotto.exceptions.LottoException
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class PurchaseSessionTest {
+    @Test
+    fun `should return a deep copy`() {
+        val session1 = PurchaseSession(amount = 1)
+        val session2 = session1.copy(manualTicketsNumber = 10)
+        assertThat(session2).isNotEqualTo(session1)
+        assertThat(session2.amount).isEqualTo(session1.amount)
+    }
+
+    @Test
+    fun `should update amount and return a new instance from the session`() {
+        val session1 = PurchaseSession()
+        val session2 = session1.updateAmount(45000)
+        assertThat(session2).isNotEqualTo(session1)
+        assertThat(session2.amount).isNotEqualTo(session1.amount)
+    }
+
+    @Test
+    fun `should throw an exception and not update amount due a invalid amount`() {
```

Comment:
```suggestion
    fun `should throw an exception and not update amount due to an invalid amount`() {
```

Small, but found a missing word 🤓